### PR TITLE
feat(dialog): create dialog

### PR DESCRIPTION
**Pull request by raushan303**
### JIRA Ticket

Jira:  [AGCT-1434](https://hello-haptik.atlassian.net/browse/AGCT-1434)

### Changelog
- Alert Dialog
- close button at top right corner
- title, subtitle and button text are passed as props  
- [Netlify](https://infallible-golick-5965d3.netlify.app/?path=/story/components-overlays-alertdialog--documentation)

### Screenshots (if appropriate):
Screenshot from 2021-08-03 13-17-43
Screenshot from 2021-08-03 17-08-59
Screenshot from 2021-08-06 10-27-28

### CODE REVIEW

**Comment by raushan303** [comment]
Netlify:- https://infallible-golick-5965d3.netlify.app/?path=/story/components-overlays-alertdialog--documentation

**Review by yashasvi13 — DISMISSED** [review]

File: src/components/Overlays/AlertDialog/AlertDialog.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+import {
+  ModalWrapper,
+  ModalOverlay,
+  Modal,
+  IconBox,
+  ButtonBox,
+} from "./styledComponents";
+
+const AlertDialog = ({
```
Comment (sidMulay) [inline src/components/Overlays/AlertDialog/AlertDialog.js:20]: Can you rename the component as `ConfirmDialog`? 

- Alert is when you want to indicate some result of an action, this seems more to me like you're taking an input from the user via the dialog.
- Alert also features success, warning and error variants

File: src/components/Overlays/AlertDialog/AlertDialog.js [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+                data-testid="primary-action"
+                themed
+                variant="primary"
+                size="regular"
+                onClick={primaryAction}
+              >
+                {primaryActionText}
+              </ButtonBox>
+            )}
+          </Box>
```
Comment (sidMulay) [inline src/components/Overlays/AlertDialog/AlertDialog.js:97]: - What happens when `primaryActionText` and `secondaryActionText`, both are not sent by the props?
- What happens user sends `primary` or `secondary` action text but does not send its action?

File: src/components/Overlays/AlertDialog/AlertDialog.js [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+                variant="body.lg"
+                color="black.haptik"
+                textAlign="center"
+                lineHeight="19px"
+              >
+                {subtitle}
+              </Text>
+            </Box>
+          )}
+          <Box justifyContent="center" marginTop="16px">
```
Comment (sidMulay) [inline src/components/Overlays/AlertDialog/AlertDialog.js:74]: This should be rendered when atleast one: primary or secondary actions are provided.

File: src/components/Overlays/AlertDialog/AlertDialog.js [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+            )}
+            {primaryActionText && (
+              <ButtonBox
+                data-testid="primary-action"
+                themed
+                variant="primary"
+                size="regular"
+                onClick={primaryAction}
+              >
+                {primaryActionText}
```
Comment (sidMulay) [inline src/components/Overlays/AlertDialog/AlertDialog.js:94]: Don't rely on user's input, set a default value to this. Use user's prop only when its present.

File: src/components/Overlays/AlertDialog/styledComponents.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+export const IconBox = styled(Icon)({
+  position: "absolute",
+  top: "15px",
+  right: "15px",
+});
+
+export const ButtonBox = styled(Button)({
+  margin: "8px",
+  padding: "12px 24px",
```
Comment (sidMulay) [inline src/components/Overlays/AlertDialog/styledComponents.js:67]: Use contracts from `space` here.
Try to use as many values as you can from the design system.

**Review by sidMulay — CHANGES_REQUESTED** [review]

File: src/components/Overlays/AlertDialog/AlertDialog.js (hunk 1 above)
Comment (raushan303) [reply src/components/Overlays/AlertDialog/AlertDialog.js]: I have renamed it with Dialog, so later maybe we can use it as alert dialog

File: src/components/Overlays/AlertDialog/AlertDialog.js (hunk 2 above)
Comment (raushan303) [reply src/components/Overlays/AlertDialog/AlertDialog.js]: now i have added default text for Yes to `primaryActionText` and No to `secondaryActionText`
have removed the secondaryAction will be using onClose only with secondaryAction
to primaryAction default empty function `() => {}` is being passed so nothing will happen if primaryAction is not passed

File: src/components/Overlays/AlertDialog/AlertDialog.js (hunk 3 above)
Comment (raushan303) [reply src/components/Overlays/AlertDialog/AlertDialog.js]: Now we have default text so this should be fine

File: src/components/Overlays/AlertDialog/AlertDialog.js (hunk 2 above)
Comment (sidMulay) [reply src/components/Overlays/AlertDialog/AlertDialog.js]: If that's the case, we should keep the `primaryAction` as a required prop I feel.

**Review by sidMulay — APPROVED** [review]

**Review by yashasvi13 — APPROVED** [review]

**Pull request merged**